src/components/Footer.tsx

```diff
@@ -0,0 +1,95 @@
+import { Dispatch, SetStateAction } from 'react';
+import cn from 'classnames';
+import { TasksFilter } from '../types/tasksFilter';
+import { Todo } from '../types/Todo';
+
+interface Props {
+  todos: Todo[],
+  tasksFilter: TasksFilter,
+  setTasksFilter: Dispatch<SetStateAction<TasksFilter>>
+  setTodos: Dispatch<SetStateAction<Todo[]>>;
+}
+
+export const Footer: React.FC<Props> = ({
+  todos,
+  tasksFilter,
+  setTasksFilter,
+  setTodos,
+}) => {
+
+   const itemsLeftCount = todos.filter((todo) => !todo.completed).length;
+   const itemsCompletedCount = todos.find((todo) => todo.completed);
+
+  const handleAll = () => {
+    setTasksFilter(TasksFilter.all);
+  };
+
+  const handleActive = () => {
+    setTasksFilter(TasksFilter.active);
+  };
+
+  const handleCompleted = () => {
+    setTasksFilter(TasksFilter.completed);
+  };
+
+  const handleClearCompleted = () => {
+    setTodos((currentTodos) => currentTodos.filter((todo) => !todo.completed));
+  };
+
+  return (
+    <footer
+      className="todoapp__footer"
+      data-cy="Footer"
+    >
+      <span className="todo-count" data-cy="TodosCounter">
+        {itemsLeftCount}
+        <span> items left</span>
```

Comment:
if 1 item text should be `1 item left` not `items`

Reply:
Ok, Ill fix that. Thanks